### PR TITLE
refactor(motion_velocity_smoother): remove unused optional

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

The optional-return used for the resample function in the smoother is no longer required since the `findClosest()` function, which was the root cause making the optional, is replaced with `findFirstNearestSegmentIndexWithSoftConstraints()`.

The debug code for the jerk_filter is also fixed in this PR since the code was broken.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by tkimura4 — APPROVED** [review]
LGTM.
(I confirmed that the Autoware worked normally on the planning simulator.)

**Comment by TakaHoribe** [comment]
@tkimura4 thank you

**Review by rej55 — APPROVED** [review]
LGTM

File: planning/motion_velocity_smoother/include/motion_velocity_smoother/resample.hpp [hunk 1]
```diff
@@ -43,12 +43,12 @@ struct ResampleParam
   double sparse_min_interval_distance;  // minimum points-interval length for sparse sampling [m]
 };
 
-boost::optional<TrajectoryPoints> resampleTrajectory(
+TrajectoryPoints resampleTrajectory(
```
Comment (wep21) [inline planning/motion_velocity_smoother/include/motion_velocity_smoother/resample.hpp:46]: @TakaHoribe I think it's better to remove unnecessary header(`boost/optional.hpp`).

Comment (TakaHoribe) [reply planning/motion_velocity_smoother/include/motion_velocity_smoother/resample.hpp:46]: Thanks! I'll take it

**Pull request merged**